extension/httpfs/httpfs.cpp

```diff
@@ -226,6 +226,11 @@ unique_ptr<HTTPResponse> HTTPFileSystem::GetRangeRequest(FileHandle &handle, str
 		    }
 		    if (static_cast<int>(response.status) < 300) { // done redirecting
 			    out_offset = 0;
+
+				if (response.HasHeader("ETag") && !hfh.etag.empty() && response.GetHeaderValue("ETag") != hfh.etag) {
+					throw HTTPException(response, "ETag was initially %s and now it returned %s, this likely means the remote file has changed.\nTry restart the read / close file-handle and restart (e.g. `DETACH` in case of database file).", hfh.etag, response.GetHeaderValue("ETag"));
```

Comment:
nit: 
```
\nTry to restart the read or close the file-handle and read the file again (e.g. `DETACH` in the file is a database file).
```

Reply:
This should probably remove the file metadata from the HTTPFS cache so that we are sure to do a new `HEAD` request upon the next time we read the file (important for when `enable_http_metadata_cache` is enabled).